src/visitors/cvode_visitor.cpp

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2023 Blue Brain Project, EPFL.
+ * See the top-level LICENSE file for details.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "visitors/cvode_visitor.hpp"
+
+#include "ast/all.hpp"
+#include "lexer/token_mapping.hpp"
+#include "pybind/pyembed.hpp"
+#include "utils/logger.hpp"
+#include "visitors/visitor_utils.hpp"
+#include <optional>
+#include <utility>
+
+namespace pywrap = nmodl::pybind_wrappers;
+
+namespace nmodl {
+namespace visitor {
+
+static int get_index(const ast::IndexedName& node) {
+    return std::stoi(to_nmodl(node.get_length()));
+}
+
+static void remove_conserve_statements(ast::StatementBlock& node) {
+    auto conserve_equations = collect_nodes(node, {ast::AstNodeType::CONSERVE});
+    if (!conserve_equations.empty()) {
+        std::unordered_set<ast::Statement*> eqs;
+        for (const auto& item: conserve_equations) {
+            eqs.insert(std::dynamic_pointer_cast<ast::Statement>(item).get());
+        }
+        node.erase_statement(eqs);
+    }
+}
+
+static std::pair<std::string, std::optional<int>> parse_independent_var(
+    std::shared_ptr<ast::Identifier> node) {
+    auto variable = std::make_pair(node->get_node_name(), std::optional<int>());
+    if (node->is_indexed_name()) {
+        variable.second = std::optional<int>(
+            get_index(*std::dynamic_pointer_cast<const ast::IndexedName>(node)));
+    }
+    return variable;
+}
+
+static std::unordered_map<std::string, int> get_name_map(const ast::Expression& node,
+                                                         const std::string& name) {
```

Comment:
What is a 'name_map'? The second argument is called `name`, but I can't figure out what's it's the name of. From the code it's an exception, i.e. if the indexed name is called `name` then don't do something. Is there a better name for this one?

Reply:
Turns out we don't need the index at all, so I changed this to a more appropriate name `get_indexed_variables`.